packages/block-library/src/table/edit.js

```diff
@@ -0,0 +1,295 @@
+
+		setAttributes( createTable( {
+			rowCount: initialRowCount,
+			columnCount: initialColumnCount,
+		} ) );
+	}
+
+	onChangeFixedLayout() {
+		const { attributes, setAttributes } = this.props;
+		const { hasFixedLayout } = attributes;
+
+		setAttributes( { hasFixedLayout: ! hasFixedLayout } );
+	}
+
+	createOnChange( { section, rowIndex, columnIndex } ) {
+		const { attributes, setAttributes } = this.props;
+
+		return ( content ) => {
+			setAttributes( updateCellContent( attributes, {
+				section,
+				rowIndex,
+				columnIndex,
+				content,
+			} ) );
+		};
+	}
+
+	createOnInsertRow( { section, rowIndex }, delta ) {
+		const { attributes, setAttributes } = this.props;
+
+		rowIndex += delta;
+
+		return () => {
+			setAttributes( insertRow( attributes, { section, rowIndex } ) );
+			this.setState( { selectedCell: null } );
+		};
+	}
+
+	createOnDeleteRow( { section, rowIndex } ) {
+		const { attributes, setAttributes } = this.props;
+
+		return () => {
+			setAttributes( deleteRow( attributes, { section, rowIndex } ) );
+			this.setState( { selectedCell: null } );
+		};
+	}
+
+	createOnInsertColumn( { section, columnIndex }, delta ) {
+		const { attributes, setAttributes } = this.props;
+
+		columnIndex += delta;
+
+		return () => {
+			setAttributes( insertColumn( attributes, { section, columnIndex } ) );
+			this.setState( { selectedCell: null } );
+		};
+	}
+
+	createOnDeleteColumn( { section, columnIndex } ) {
+		const { attributes, setAttributes } = this.props;
+
+		return () => {
+			setAttributes( deleteColumn( attributes, { section, columnIndex } ) );
+			this.setState( { selectedCell: null } );
+		};
+	}
+
+	createOnFocus( selectedCell ) {
+		return () => {
+			this.setState( { selectedCell } );
+		};
+	}
+
+	getTableControls() {
+		const { selectedCell } = this.state;
+
+		return [
+			{
+				icon: 'table-row-before',
+				title: __( 'Add Row Before' ),
+				isDisabled: ! selectedCell,
+				onClick: selectedCell && this.createOnInsertRow( selectedCell ),
+			},
+			{
+				icon: 'table-row-after',
+				title: __( 'Add Row After' ),
+				isDisabled: ! selectedCell,
+				onClick: selectedCell && this.createOnInsertRow( selectedCell, 1 ),
+			},
+			{
+				icon: 'table-row-delete',
+				title: __( 'Delete Row' ),
+				isDisabled: ! selectedCell,
+				onClick: selectedCell && this.createOnDeleteRow( selectedCell ),
+			},
+			{
+				icon: 'table-col-before',
+				title: __( 'Add Column Before' ),
+				isDisabled: ! selectedCell,
+				onClick: selectedCell && this.createOnInsertColumn( selectedCell ),
+			},
+			{
+				icon: 'table-col-after',
+				title: __( 'Add Column After' ),
+				isDisabled: ! selectedCell,
+				onClick: selectedCell && this.createOnInsertColumn( selectedCell, 1 ),
+			},
+			{
+				icon: 'table-col-delete',
+				title: __( 'Delete Column' ),
+				isDisabled: ! selectedCell,
+				onClick: selectedCell && this.createOnDeleteColumn( selectedCell ),
+			},
+		];
+	}
+
+	renderSection( { type, rows } ) {
```

Comment:
Since it contains a `RichText` that needs to update attributes, I think it make sense to have a context. Otherwise we can pass `attributes` and `setAttributes` as wel. I don't mind either way...

Reply:
I also think that it would be a good idea to extract this method to its own file and promote it to an independent component. It will decrease the file size of the current `edit` implementation and will help to better understand which props are used by those `Section` components.